cli/options.go

```diff
@@ -251,9 +250,6 @@ func GetEnvFromFile(currentEnv map[string]string, workingDir string, filename st
 		return envMap, err
 	}
 	for k, v := range env {
-		if _, set := currentEnv[k]; set {
-			continue
-		}
```

Comment:
No. That's exactly the intention.
If the users want to choose the OS one they can use something like this in the `.env` file:
```
MYVAR=${MYVAR:-DefaultValue}
```
This would use the OS variable when present and use `DefaultValue` when unset or empty.

That's why I had to fix the [default values](https://github.com/compose-spec/compose-go/pull/276) first.

Reply:
Ahhh okay I just did some sleuthing, and v1 _does_ behave in the same way that you are making the changes here: env has higher precedence than shell, so interpolation/default syntax is required for any values you want to "fall through" to shell.

HOWEVER, my confusion came from the fact that the Compose docs, even historically going back to v1, actually state the opposite behavior 🙈  (e.g. here's the page in 2019 -> https://github.com/docker/docker.github.io/blob/f2093d4e2a5f950d1fd8e6dc61441a1d90cecdb4/compose/environment-variables.md?plain=1#L115-L117)

So, I think for v1 compat, this makes sense! But we'll definitely need to take care to find any remaining errant references to the order being the other way around, and given that v2 is now flip-flopping its behavior, should make sure we call that out explicitly in release notes, along with the "workaround" (use `MYVAR=${MYVAR:-DefaultValue}` syntax in `.env` file), and explanation that it should now behave the same between v1+v2 😅